modules/client/shared/src/main/scala/lucuma/itc/client/ItcResult.scala

```diff
@@ -16,65 +16,58 @@ import lucuma.core.util.TimeSpan
 
 import java.math.MathContext
 
-sealed trait ItcResult {
-
-  def resultType: String =
-    this match {
-      case ItcResult.Error(_)         => ItcResult.Error.ResultType
-      case ItcResult.Success(_, _, _) => ItcResult.Success.ResultType
-    }
-
-  def toEither: Either[ItcResult.Error, ItcResult.Success] =
-    this match {
-      case e: ItcResult.Error   => e.asLeft[ItcResult.Success]
-      case s: ItcResult.Success => s.asRight[ItcResult.Error]
-    }
-
-  def error: Option[ItcResult.Error] =
-    toEither.swap.toOption
-
-  def success: Option[ItcResult.Success] =
-    toEither.toOption
-
-}
+sealed trait ItcResult
 
 object ItcResult {
 
   given Decoder[ItcResult] with
     def apply(c: HCursor): Decoder.Result[ItcResult] =
-      c.downField("resultType").as[String].flatMap {
-        case "Success" => c.as[Success].widen[ItcResult]
-        case "Error"   => c.as[Error].widen[ItcResult]
-        case rt        =>
+      c.downField("__typename").as[String].flatMap {
+        case "ExposureEstimate" => c.as[Success].widen[ItcResult]
+        case "SourceTooBright"  => c.as[SourceTooBright].widen[ItcResult]
+        case "CalculationError" => c.as[Error].widen[ItcResult]
```

Comment:
Should we make the Scala type names match the Schema type names?  `Success` => `ExposureTimeEstimate`, `Error` => `CalculationError`?